server/src/main/java/io/crate/planner/operators/HashJoin.java

```diff
@@ -98,23 +98,27 @@ public ExecutionPlan build(DependencyCarrier executor,
         var lhsHashSymbols = new ArrayList<Symbol>();
         var rightRelationNames = rhs.getRelationNames();
         var leftRelationNames = lhs.getRelationNames();
-        for (var entry : hashSymbols.entrySet()) {
-            var relationName = entry.getKey();
-            var symbols = entry.getValue();
-            if (symbols == null) {
-                continue;
-            }
-            if (rightRelationNames.contains(relationName)) {
-                for (Symbol symbol : symbols) {
-                    rhsHashSymbols.add(paramBinder.apply(symbol));
+        for (var condition : QuerySplitter.split(joinCondition).entrySet()) {
```

Comment:
This seems to work correctly, but we could move the whole hash symbol generation logic into HashJoinConditionSymbolsExtractor which generates the correct set for lhs/rhs.

Reply:
I don't have a preference really, it's also fine to keep the for loop here, and have `extract()` operate on a single condition.

Reply:
The purpose of the `HashJoinConditionSymbolsExtractor` is to extract the hash-symbols from the join-condition for the hash-join and the hash-symbols are not extracted correctly. So instead of building logic around to make it work, I think it would be better to fixthe `HashJoinConditionSymbolsExtractor` itself. It also makes it also easier to test.

Reply:
ok, but the name is `ConditionSymbolsExtractor` so it should take a single join condition and extract symbols, that's why I find it ok as is, but if you feel it should do everything also totally fine.